src/gui/preferences.c

```diff
@@ -564,157 +582,163 @@ static void cairo_destroy_from_pixbuf(guchar *pixels, gpointer data)
+    focal_length = g_strdup("");
+    smaker = g_strdup("");
+    smodel = g_strdup("");
+    slens = g_strdup("");
+  }
+  else
+  {
+    smaker = g_strdup(maker);
+    smodel = g_strdup(model);
+    slens = g_strdup(lens);
 
-      if(iso_min == 0.0 && iso_max == FLT_MAX)
-        iso = g_strdup("%");
-      else
-        iso = g_strdup_printf("%zu – %zu", (size_t)iso_min, (size_t)iso_max);
-
-      for(min = 0; min < dt_gui_presets_exposure_value_cnt && exposure_min > dt_gui_presets_exposure_value[min]; min++)
-        ;
-      for(max = 0; max < dt_gui_presets_exposure_value_cnt && exposure_max > dt_gui_presets_exposure_value[max]; max++)
-        ;
-      if(min == 0 && max == dt_gui_presets_exposure_value_cnt - 1)
-        exposure = g_strdup("%");
-      else
-        exposure = g_strdup_printf("%s – %s", dt_gui_presets_exposure_value_str[min],
-                                   dt_gui_presets_exposure_value_str[max]);
-
-      for(min = 0; min < dt_gui_presets_aperture_value_cnt && aperture_min > dt_gui_presets_aperture_value[min]; min++)
-        ;
-      for(max = 0; max < dt_gui_presets_aperture_value_cnt && aperture_max > dt_gui_presets_aperture_value[max]; max++)
-        ;
-      if(min == 0 && max == dt_gui_presets_aperture_value_cnt - 1)
-        aperture = g_strdup("%");
-      else
-        aperture = g_strdup_printf("%s – %s", dt_gui_presets_aperture_value_str[min],
-                                   dt_gui_presets_aperture_value_str[max]);
+    if(iso_min == 0.0 && iso_max == FLT_MAX)
+      iso = g_strdup("%");
+    else
+      iso = g_strdup_printf("%zu – %zu", (size_t)iso_min, (size_t)iso_max);
+
+    for(min = 0; min < dt_gui_presets_exposure_value_cnt && exposure_min > dt_gui_presets_exposure_value[min]; min++)
+      ;
+    for(max = 0; max < dt_gui_presets_exposure_value_cnt && exposure_max > dt_gui_presets_exposure_value[max]; max++)
+      ;
+    if(min == 0 && max == dt_gui_presets_exposure_value_cnt - 1)
+      exposure = g_strdup("%");
+    else
+      exposure = g_strdup_printf("%s – %s", dt_gui_presets_exposure_value_str[min],
+                                 dt_gui_presets_exposure_value_str[max]);
+
+    for(min = 0; min < dt_gui_presets_aperture_value_cnt && aperture_min > dt_gui_presets_aperture_value[min]; min++)
+      ;
+    for(max = 0; max < dt_gui_presets_aperture_value_cnt && aperture_max > dt_gui_presets_aperture_value[max]; max++)
+      ;
+    if(min == 0 && max == dt_gui_presets_aperture_value_cnt - 1)
+      aperture = g_strdup("%");
+    else
+      aperture = g_strdup_printf("%s – %s", dt_gui_presets_aperture_value_str[min],
+                                 dt_gui_presets_aperture_value_str[max]);
 
-      if(focal_length_min == 0.0 && focal_length_max == 1000.0)
-        focal_length = g_strdup("%");
-      else
-        focal_length = g_strdup_printf("%d – %d", focal_length_min, focal_length_max);
-    }
+    if(focal_length_min == 0.0 && focal_length_max == 1000.0)
```

Comment:
Here we compare `focal_length_min` & `focal_length_max` to float, but those are integers.

So maybe it should be:

`if(focal_length_min == 0.0f && focal_length_max == 1000.0f)`

and the variable float (see below).

Reply:
Your suggestion is correct, but "0.0" parses as a double in C.